libs/api/domains/license-service/src/lib/mappers/disabilityLicenseMapper.ts

```diff
@@ -1,63 +1,90 @@
 
     const typedPayload = payload as Array<OrorkuSkirteini>
 
-    const label = labels?.labels
-    const mappedPayload: Array<GenericUserLicensePayload> = typedPayload.map(
-      (t) => {
+    const { formatMessage } = await this.intlService.useIntl(
+      [LICENSE_NAMESPACE],
+      locale,
+    )
+
+    const mappedPayload: Array<GenericLicenseMappedPayloadResponse> =
+      typedPayload.map((t) => {
+        const isExpired: boolean | undefined = t.gildirtil
+          ? !isAfter(new Date(t.gildirtil), new Date())
+          : undefined
+
         const data: Array<GenericLicenseDataField> = [
           {
             type: GenericLicenseDataFieldType.Value,
-            name: 'Grunnupplýsingar örorkuskírteinis',
-            label: label ? label['fullName'] : i18n.fullName[locale],
+            name: formatMessage(m.basicInfoDisabilityLicense),
+            label: formatMessage(m.fullName),
             value: t.nafn ?? '',
           },
           {
             type: GenericLicenseDataFieldType.Value,
-            label: label ? label['publisher'] : i18n.publisher[locale],
+            label: formatMessage(m.publisher),
             value: 'Tryggingastofnun',
           },
           {
             type: GenericLicenseDataFieldType.Value,
-            label: label ? label['validTo'] : i18n.validTo[locale],
-            value: t.gildirtil?.toISOString() ?? '',
+            label: formatMessage(m.validTo),
+            value: t.gildirtil ? formatDate(t.gildirtil) : '',
+            tag:
+              isExpired !== undefined && t.gildirtil
+                ? expiryTag(formatMessage, isExpired)
+                : undefined,
           },
         ]
 
         return {
-          data,
-          rawData: JSON.stringify(t),
-          metadata: {
-            licenseNumber: t.kennitala?.toString() ?? '',
-            licenseId: DEFAULT_LICENSE_ID,
-            expired: t.gildirtil
-              ? !isAfter(new Date(t.gildirtil), new Date())
-              : null,
+          licenseName: formatMessage(m.disabilityCard),
+          type: 'user',
+          payload: {
+            data,
+            rawData: JSON.stringify(t),
+            metadata: {
+              licenseNumber: t.kennitala?.toString() ?? '',
+              subtitle: formatMessage(m.licenseNumberVariant, {
+                arg: t.kennitala?.toString() ?? formatMessage(m.unknown),
+              }),
+              licenseId: DEFAULT_LICENSE_ID,
+              expired: isExpired,
+              expireDate: t.gildirtil?.toISOString() ?? undefined,
+              displayTag: expiryTag(formatMessage, isExpired),
+              name: formatMessage(m.disabilityCard),
+              title: formatMessage(m.yourDisabilityLicense),
+              description: [
+                { text: formatMessage(m.yourDisabilityLicenseDescription) },
+              ],
+            },
           },
         }
-      },
-    )
+      })
 
     return mappedPayload
```

Comment:
Do the changes of this return type (and in other license mappers) have any effect of the return type of the endpoint itself? Just to make sure this is 100% backwards compatible 😄

Reply:
It shouldn't, the resolver still returns the same type.